btrdb/experimental/bytewax_connectors.py

```diff
@@ -0,0 +1,168 @@
+            def background_worker(self):
+                try:
+                    # Avoid exessive weakref lookups
+                    # by doing the lookup upfront initially.
+                    db = self._db
+                    del_event = self._del_event
+                    selector_fn = self._selector_fn
+                    worker_index = self._worker_index
+                    worker_count = self._worker_count
+                    data_queue = self._data_queue
+                    heartbeat_interval = self._heartbeat_interval
+                    last_heartbeat = time.monotonic()
+                    selector_refresh_interval = self._selector_refresh_interval
+                    last_selector_refresh = time.monotonic() - selector_refresh_interval
+                    current_uuids = set()
+                    while True:
+                        now = time.monotonic()
+                        if (now - last_selector_refresh) >= selector_refresh_interval:
+                            last_selector_refresh = now
+                            next_uuids = {
+                                uuid
+                                for uuid in selector_fn(db)
+                                if uuid.int % worker_count == worker_index
+                            }
+                            added_uuids = next_uuids - current_uuids
+                            removed_uuids = current_uuids - next_uuids
+                            if len(added_uuids) != 0 or len(removed_uuids) != 0:
+                                self._update_queue.put([added_uuids, removed_uuids])
+                            current_uuids = next_uuids
+                        if (
+                            heartbeat_interval is not None
+                            and (now - last_heartbeat) >= heartbeat_interval
+                        ):
+                            last_heartbeat = now
+                            for uuid in current_uuids:
+                                while not del_event.is_set():
+                                    try:
+                                        data_queue.put((uuid, _empty_values), 0.1)
+                                        break
+                                    except queue.Full:
+                                        pass
+                        if del_event.wait(1.0):
+                            return
+                except Exception as e:
+                    self._background_worker_exception = e
+
+            weakself = weakref.proxy(self)
+            data = db.ep.subscribe(self._update_queue)
+            self._background_worker = threading.Thread(
+                target=background_worker, args=[weakself], daemon=True
+            )
+            self._read_worker = threading.Thread(
+                target=read_worker, args=[weakself, data], daemon=True
+            )
+            self._background_worker.start()
+            self._read_worker.start()
+
+        def next(self):
```

Comment:
I attempted to use the bytewax flow example, and it seems in bytewax=='0.17.1' there is no `next` it's either `next_batch` or `next_awake`

Reply:
Yeah, this was written against v0.16, needs to be reworked slightly for 0.17.